site/docs/relations/physical_relations.md

```diff
@@ -71,6 +71,30 @@ The merge equijoin does a join by taking advantage of two sets that are sorted o
 | Post Join Predicate | An additional expression that can be used to reduce the output of the join operation post the equality condition. Minimizes the overhead of secondary join conditions that cannot be evaluated using the equijoin keys. | Optional, defaults true.    |
 | Join Type           | One of the join types defined in the Join operator.                                                                                                                                                                     | Required                    |
 
+
+
+## Mark Join Operator
+
+A mark join utilizes a previously applied mark to greatly reduce the input to be processed.
+
+| Signature            | Value                                                        |
+| -------------------- | ------------------------------------------------------------ |
+| Inputs               | 2                                                            |
+| Outputs              | 1                                                            |
+| Property Maintenance | Distribution is maintained. Orderedness is eliminated.       |
+| Direct Output Order  | Same as the [Join](logical_relations.md#join-operator) operator. |
+
+### Mark Join Properties
+
+| Property        | Description                                                                                                                                                                         | Required                                                    |
+|-----------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------|
+| Left Input      | A relational input.                                                                                                                                                                 | Required                                                    |
+| Right Input     | A relational input.                                                                                                                                                                 | Required                                                    |
+| Mark Reference  | A nullable boolean field reference that is used to filter the right input.  If the mark is null, the row is not included in the join.                                               | Required.                                                   |
```

Comment:
That's it?  Why not just use a `FilterRel`?

Reply:
This description is not correct - a `mark` join does not change the cardinality of the left table (much like a `single` join). The marker is either `true`, `false` or `NULL` depending on if a match is found and/or if any `NULL` values were encountered. Often the marker is used in a filter later on - but this is not required. If the marker is exclusively used in a filter the join can also be converted into a semi join and the marker is not necessary. It's described in [this paper](http://btw2017.informatik.uni-stuttgart.de/slidesandpapers/F1-10-37/paper_web.pdf).

Reply:
I've updated the documentation.  It hopefully is more clear what the mark join is trying to accomplish.

I've also removed the mark reference as that appears to be an internal detail for the mark join (an internal hash table is constructed instead of the right join input having a new column introduced).